offscreen/scripts/ledger.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  LedgerAction,
+  OffscreenCommunicationEvents,
+  OffscreenCommunicationTarget,
+  KnownOrigins,
+} from '../../shared/constants/offscreen-communication';
+import { CallbackProcessor } from './callback-processor';
+
+const LEDGER_FRAME_TARGET = 'LEDGER-IFRAME';
+
+/**
+ * The ledger keyring iframe will send this event name when the ledger is
+ * connected to the iframe.
+ */
+const LEDGER_KEYRING_IFRAME_CONNECTED_EVENT = 'ledger-connection-event';
+
+const callbackProcessor = new CallbackProcessor();
+
+export default function init() {
+  const iframe = document.querySelector('iframe');
+  // This listener receives action responses from the live ledger iframe
+  // Then forwards the response to the offscreen bridge
+  window.addEventListener('message', ({ origin, data, source }) => {
+    if (origin !== KnownOrigins.ledger || source !== iframe?.contentWindow) {
```

Comment:
this pattern we formed back then of checking both the origin and the sender's iframe is great - can we please apply this to the rest of the integrations as well?

Reply:
I'm unsure what you mean. This is the only integration that uses an iframe?

Reply:
Who's `event.source` [here](https://github.com/MetaMask/metamask-extension/pull/22914/files#diff-8478186b287730ee76bf0f0ee62e6e5428a63b26d058a32af31b35563cbc984eR55)? Are we the ones forming that window? If so, can we similarly verify `event.source` === that window?

Reply:
We open a window to the lattice connector used to enter in the device security code and confirm that the event is coming from that known origin. I have added a check there to see if event.source matches browserTab (the window we just opened) but i'm going to need to find someone with a lattice device now to test that it still works appropriately cause I do not have one

Reply:
@weizman success! I added a check for the event.source like you suggested and @montelaidev checked and it works